debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/TDumpReader.java

```diff
@@ -411,30 +418,58 @@ public int getBytesAt(long address, byte[] buffer, int offset,
 				int length) throws MemoryFault
 		{
 			try {
-				addressSpace.read(address, buffer, offset, length);
+				if (PatchedMemoryList.isPatched(address, length) == false) {
+					addressSpace.read(address, buffer, offset, length);
+					return length;
+				}
+				List<MemoryRecord> list = PatchedMemoryList.getPatched(address, length);
+				if (list == null) {
+					return getBytesAt(address, buffer, 0, buffer.length);
+				}
+				int cursor = offset;
+				for (MemoryRecord mr : list) {
+					byte [] content = mr.getContent();
+					if (content == null) {
+						content = new byte [mr.getLength()];
+						addressSpace.read(mr.getStart(), content, 0, content.length);
+					}
+					System.arraycopy(content, 0, buffer, cursor, mr.getLength());
+					cursor += mr.getLength();
+				}
+				return length;
 			} catch (IOException e) {
 				throw new MemoryFault(address, e);
 			}
-			
-			return length;
 		}
 
 		public int getIntAt(long address) throws MemoryFault
 		{
 			try {
-				return addressSpace.readInt(address);
+				if (PatchedMemoryList.isPatched(address, 4) == false) {
+					return addressSpace.readInt(address);
+				}				
 			} catch (IOException e) {
 				throw new MemoryFault(address, e);
 			}
+			
+			byte [] buffer = new byte[4];
+			getBytesAt(address, buffer);
+			return ByteBuffer.wrap(buffer).getInt();
```

Comment:
This should either explicitly set the byte-order of the buffer or include a comment why the default (big-endian) is appropriate.